src/authority.rs

```diff
@@ -773,7 +773,7 @@ pub enum Host<'host> {
     IPv4Address(Ipv4Addr),
 
     /// An IPv6 address. This will always be encased in brackets (`'['` and `']'`).
-    IPv6Address(Ipv6Addr),
+    IPv6Address(Ipv6Addr, Cow<'host, str>),
```

Comment:
I generally prefer using strong types rather than strings (as you can see by the library having `Port`, `Username`, `Password`, etc. types). I'd prefer if this was a new `ZoneID` type.